src/IssueViz/Selection/SelectedVisualizationValidityMonitor.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Selection
+{
+    internal interface ISelectedVisualizationValidityMonitor : IDisposable
```

Comment:
The interface/class needs a doc comment explaining what it's responsible for. It's not obvious without reading all of the code.

Reply:
I was trying to think of a comment that describes what this class is for, and in fact it seems it just exists to patch an incorrect selection status that could be created by the interaction of all of the other classes. That makes it seem like a design smell in its current form (hence my other comment about selection overall needing more work). Hopefully this class will disappear at some point if we can simplify the rest of the selection logic.